### DISCUSSION TITLE
How can I automatically create a table of contents for my index.md files?

### DISCUSSION THREAD

**Discussion by ghost** [squidfunk/mkdocs-material#7000] [Q&A] [answered]
Hello everyone!

I have been trying to solve this problem following the advice given in this forum (https://github.com/mkdocs/mkdocs/issues/466), but so far nothing has worked. In my **index.md** file, I have tried adding **[TOC]** as suggested, but it hasn't helped. Is there a quick solution to this issue that I can try?

**Thanks!**

**Comment by alexvoss** [comment]
I assume you are talking about a TOC for the current page? That is implemented in the [Table of Contents Extension](https://python-markdown.github.io/extensions/toc/) or Python Markdown, which you would need to turn on in your `mkdocs.yml`:

```
markdown_extensions:
  - toc
```

Once this is done, the `[TOC]` placeholder should be replaced with a table of contents.

**Reply by ghost** [reply]
**Thank you for your response! I am very grateful for your assistance.**

It turns out that I had asked the question in a way that was incorrect. I wanted the automatically-generated **navigation** on the left-hand side to appear inside **index.md.** For example (https://www.mkdocs.org/user-guide/)

**Reply by alexvoss** [reply]
The example you point to is actually [using Markdown to statically define that navigation](https://raw.githubusercontent.com/mkdocs/mkdocs/master/docs/user-guide/README.md). 

Automatically adding this to your pages might require a bit of effort in the form of  a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) to produce. In the [on_page_markdown](https://www.mkdocs.org/dev-guide/plugins/#on_page_markdown) event you can search the placeholder for the ToC and replace it with a list you generate. You'd have to see if you can access the navigation definition from there somewhere and figure out which part you need to render on each page. 

I do wonder, though, what it is you are trying to achieve for your users and if it is worth the effort? One of the nice things about Material is that it handles navigation well. What would be the benefit of augmenting or replacing it?

**Reply by ghost** [reply]
I'm trying to customize the navigation so that when the website opens, only the main sections are visible, and when users visit the site, the navigation appears in more detail.

**Reply by alexvoss** [reply, marked as answer]
That is what you get by default when you define sections in your navigation. They are collapsed and only the section the user is in opens. You can also [lift the main sections up into tabs](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#navigation-tabs-with-tabs) and [hide the sidebars on your homepage](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#hiding-the-sidebars). 

It is up to you but I do wonder if configuration as described above is not sufficient. It certainly seems a lot simpler to achieve.

**Reply by ghost** [reply]
I hadn't considered that. Thank you so much, kind sir. I am really grateful that you are here to help. Have a great day and enjoy yourself!
